bigtop-packages/src/common/presto/install_presto.sh

```diff
@@ -0,0 +1,209 @@
+     --lib-dir=DIR               path to install Presto home [/usr/lib/presto]
+     --installed-lib-dir=DIR     path where lib-dir will end up on target system
+     --bin-dir=DIR               path to install bins [/usr/bin]
+     ... [ see source for more similar options ]
+  "
+  exit 1
+}
+
+OPTS=$(getopt \
+  -n $0 \
+  -o '' \
+  -l 'prefix:' \
+  -l 'doc-dir:' \
+  -l 'lib-dir:' \
+  -l 'installed-lib-dir:' \
+  -l 'bin-dir:' \
+  -l 'source-dir:' \
+  -l 'build-dir:' -- "$@")
+
+if [ $? != 0 ] ; then
+    usage
+fi
+
+eval set -- "$OPTS"
+while true ; do
+    case "$1" in
+        --prefix)
+        PREFIX=$2 ; shift 2
+        ;;
+        --build-dir)
+        BUILD_DIR=$2 ; shift 2
+        ;;
+        --source-dir)
+        SOURCE_DIR=$2 ; shift 2
+        ;;
+        --doc-dir)
+        DOC_DIR=$2 ; shift 2
+        ;;
+        --lib-dir)
+        LIB_DIR=$2 ; shift 2
+        ;;
+        --installed-lib-dir)
+        INSTALLED_LIB_DIR=$2 ; shift 2
+        ;;
+        --bin-dir)
+        BIN_DIR=$2 ; shift 2
+        ;;
+        --)
+        shift ; break
+        ;;
+        *)
+        echo "Unknown option: $1"
+        usage
+        exit 1
+        ;;
+    esac
+done
+
+for var in PREFIX BUILD_DIR SOURCE_DIR; do
+  if [ -z "$(eval "echo \$$var")" ]; then
+    echo Missing param: $var
+    usage
+  fi
+done
+
+if [ -f "$SOURCE_DIR/bigtop.bom" ]; then
+  . $SOURCE_DIR/bigtop.bom
+fi
+
+MAN_DIR=${MAN_DIR:-/usr/share/man/man1}
+DOC_DIR=${DOC_DIR:-/usr/share/doc/presto}
+LIB_DIR=${LIB_DIR:-/usr/lib/presto}
+VAR_DIR=${VAR_DIR:-/var/lib/presto}
+LOG_DIR=${LOG_DIR:-/var/log/presto}
+RUN_DIR=${RUN_DIR:-/var/run/presto}
+INSTALLED_LIB_DIR=${INSTALLED_LIB_DIR:-/usr/lib/presto}
+BIN_DIR=${BIN_DIR:-/usr/bin}
+CONF_DIR=${CONF_DIR:-/etc/presto}
+CONF_DIST_DIR=${CONF_DIST_DIR:-/etc/presto.dist}
+DEFAULT_DIR=${DEFAULT_DIR:-/etc/default}
+
+install -d -m 0755 $PREFIX/$CONF_DIST_DIR
+install -d -m 0755 $PREFIX/$LIB_DIR
+install -d -m 0755 $PREFIX/$LIB_DIR/lib
+install -d -m 0755 $PREFIX/$LIB_DIR/bin
+install -d -m 0755 $PREFIX/$LIB_DIR/var
+install -d -m 0755 $PREFIX/$LIB_DIR/plugin
+install -d -m 0755 $PREFIX/$DOC_DIR
+install -d -m 0755 $PREFIX/$VAR_DIR
+install -d -m 0755 $PREFIX/$LOG_DIR
+install -d -m 0755 $PREFIX/$RUN_DIR
+install -d -m 0755 $PREFIX/$DEFAULT_DIR
+
+cp -ra ${BUILD_DIR}/* $PREFIX/$LIB_DIR/
+mv $PREFIX/$LIB_DIR/docs/* $PREFIX/$DOC_DIR/
+
+chmod +x $PREFIX/$LIB_DIR/bin/presto
+
+install -d -m 0755 $PREFIX/$CONF_DIST_DIR
+install -d -m 0755 $PREFIX/$CONF_DIST_DIR/catalog
+
+cat > $PREFIX/$CONF_DIST_DIR/node.properties <<EOF
+node.environment=production
+node.id=ffffffff-ffff-ffff-ffff-ffffffffffff
+node.data-dir=/var/lib/presto/
+EOF
+
+cat > $PREFIX/$CONF_DIST_DIR/jvm.config <<EOF
+-server
+-Xmx1G
+-XX:+UseG1GC
+-XX:G1HeapRegionSize=32M
+-XX:+UseGCOverheadLimit
+-XX:+ExplicitGCInvokesConcurrent
+-XX:+HeapDumpOnOutOfMemoryError
+-XX:+ExitOnOutOfMemoryError
+-DHADOOP_USER_NAME=hive
```

Comment:
Is it always should be hive?

Reply:
'hive' is a HDFS username for Presto server. if we specify the username, Presto server will access HDFS as user of presto process. IMHO, if users don't use Kerberos, 'hive' is appropriate one. 

https://prestodb.io/docs/current/connector/hive.html